pkg/asset/installconfig/ibmcloud/metadata.go

```diff
@@ -131,8 +132,8 @@ func (m *Metadata) ControlPlaneSubnets(ctx context.Context) (map[string]Subnet,
 
 // Client returns a client used for making API calls to IBM Cloud services.
 func (m *Metadata) Client() (*Client, error) {
-	m.mutex.Lock()
-	defer m.mutex.Unlock()
```

Comment:
I think this looks good, and am ready to approve it. However, I have one observation I'd like to throw out there...

If m.client is not null, from line 137, then is there value in locking the mutex at all? I'm assuming the reason for doing so is to handle the NewClient() and/or client.SetVPCServiceURLForRegion() calls. In the case where those are not being called because the client already exists, perhaps it's worth not locking at all?
```
if m.client != nil {
        return m.client, nil
}

m.clientMutex.Lock()
defer m.clientMutex.Unlock()

client,err := NewClient()
...
m.client = client
return m.client, nil
```